# Objective of Analysis (Markdown)
Analisis ini bertujuan untuk mengidentifikasi segmen customer dengan tingkat churn tertinggi sebagai dasar penentuan prioritas penanganan oleh tim manajemen. Fokus analisis berada pada perbandingan churn antar segmen, bukan pada identifikasi penyebab churn secara mendalam

In [1]:
import pandas as pd

url = "https://raw.githubusercontent.com/AlvitoDwiP/churn-analysis/main/churn/data/raw/WA_Fn-UseC_-Telco-Customer-Churn.csv"
df = pd.read_csv(url)


## Overal Churn Baseline 
Distribusi churn secara keseluruhan digunakan sebagai baseline untuk membandingkan tingkat churn pada masing-masing segmen customer.

In [2]:
df['Churn'].value_counts(normalize=True)


Churn
No     0.73463
Yes    0.26537
Name: proportion, dtype: float64

In [3]:
churn_contract = (
    df.groupby('Contract')['Churn']
      .value_counts(normalize=True)
      .unstack()
      .sort_values('Yes', ascending=False)
)

churn_contract


Churn                 No       Yes
Contract                          
Month-to-month  0.572903  0.427097
One year        0.887305  0.112695
Two year        0.971681  0.028319

Analisis ini menunjukkan perbedaan tingkat churn antar jenis kontrak (Contract), yang membantu mengidentifikasi segmen dengan risiko churn tertinggi.

In [4]:
churn_internet = (
    df.groupby('InternetService')['Churn']
      .value_counts(normalize=True)
      .unstack()
      .sort_values('Yes', ascending=False)
)

churn_internet


Churn                  No       Yes
InternetService                    
Fiber optic      0.581072  0.418928
DSL              0.810409  0.189591
No               0.925950  0.074050

Analisis ini menunjukkan perbedaan tingkat churn antar jenis internet (InternetService), yang membantu mengidentifikasi segmen dengan risiko churn tertinggi.

In [5]:
churn_senior = (
    df.groupby('SeniorCitizen')['Churn']
      .value_counts(normalize=True)
      .unstack()
)

churn_senior


Churn                No       Yes
SeniorCitizen                    
0              0.763938  0.236062
1              0.583187  0.416813

Analisis ini menunjukkan perbedaan tingkat churn antar senior citizen yang dimana angka 1 adalah senior dan 0 adalah non-senior

## Key Segment-level Findings

### Fakta
- Churn tertinggi muncul pada Month-to-month (42.7%)
- Fiber optic (41.9%), dan Senior citizen (41.7%).

#### Perbandingan
- Month-to-month (42.7%) jauh lebih tinggi dibanding One year 11.3% dan Two year 2.8%
- Fiber optic (41.9%) lebih tinggi dibanding DSL 19.0% dan No internet 7.4% 
- Senior citizen (41.7%) lebih tinggi dibanding non‑senior (23.6%).

### Implikasi
Churn terkonsentrasi pada segmen kontrak jangka pendek, layanan fiber optic, dan customer senior, sehingga segmen-segmen ini memberi dampak terbesar terhadap stabilitas retensi dan pendapatan.